code/ui/components/src/components/Button/Button.tsx

```diff
@@ -1,272 +1,216 @@
-            background: darken(0.05, color),
-          },
-          '&:active': {
-            boxShadow: 'rgba(0, 0, 0, 0.1) 0 0 0 3em inset',
-          },
-          '&:focus': {
-            boxShadow: `${rgba(color, 1)} 0 1px 9px 2px`,
-            outline: 'none',
-          },
-          '&:focus:hover': {
-            boxShadow: `${rgba(color, 0.2)} 0 8px 18px 0px`,
-          },
-        }
-      : {};
-  },
-  ({ theme, tertiary, inForm, small }) =>
-    tertiary
-      ? {
-          background: theme.button.background,
-          color: theme.input.color,
-          boxShadow: `${theme.button.border} 0 0 0 1px inset`,
-          borderRadius: theme.input.borderRadius,
-
-          ...(inForm && small ? { padding: '10px 16px' } : {}),
-
-          '&:hover': {
-            background:
-              theme.base === 'light'
-                ? darken(0.02, theme.button.background)
-                : lighten(0.03, theme.button.background),
-            ...(inForm
-              ? {}
-              : {
-                  boxShadow: 'rgba(0,0,0,.2) 0 2px 6px 0, rgba(0,0,0,.1) 0 0 0 1px inset',
-                }),
-          },
-          '&:active': {
-            background: theme.button.background,
-          },
-          '&:focus': {
-            boxShadow: `${rgba(theme.color.secondary, 1)} 0 0 0 1px inset`,
-            outline: 'none',
-          },
-        }
-      : {},
-  ({ theme, outline }) =>
-    outline
-      ? {
-          boxShadow: `${transparentize(0.8, theme.color.defaultText)} 0 0 0 1px inset`,
-          color: transparentize(0.3, theme.color.defaultText),
-          background: 'transparent',
-
-          '&:hover, &:focus': {
-            boxShadow: `${transparentize(0.5, theme.color.defaultText)} 0 0 0 1px inset`,
-            outline: 'none',
-          },
+    // Match the old API with the new API
+    // TODO: Remove this after 9.0
+    if (props.secondary || props.tertiary || props.gray || props.outline || props.inForm) {
+      lovalVariant = 'outline';
+      localSize = 'medium';
+    }
 
-          '&:active': {
-            boxShadow: `${transparentize(0.5, theme.color.defaultText)} 0 0 0 2px inset`,
-            color: transparentize(0, theme.color.defaultText),
-          },
-        }
-      : {},
-  ({ theme, outline, primary }) => {
-    const color = theme.color.primary;
+    if (
+      props.small ||
+      props.isLink ||
+      props.primary ||
+      props.secondary ||
+      props.tertiary ||
+      props.gray ||
+      props.outline ||
+      props.inForm ||
+      props.containsIcon
+    ) {
+      const buttonContent = React.Children.toArray(props.children).filter(
+        (e) => typeof e === 'string' && e !== ''
+      );
 
-    return outline && primary
-      ? {
-          boxShadow: `${color} 0 0 0 1px inset`,
-          color,
+      deprecate(
+        `Use of deprecated props in the button ${
+          buttonContent.length > 0 ? `"${buttonContent[0]}"` : 'component'
```

Comment:
would this work too?
```suggestion
          buttonContent.length > 0 ? `"${buttonContent.join(' ')}"` : 'component'
```

Reply:
Yes. I committed the change